samples/snippets/src/test/java/com/example/spanner/EncryptionKeyIT.java

```diff
@@ -0,0 +1,135 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.spanner;
+
+import static com.google.common.truth.Truth.assertThat;
+
+import com.google.cloud.spanner.DatabaseAdminClient;
+import com.google.cloud.spanner.Spanner;
+import com.google.cloud.spanner.SpannerOptions;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+/**
+ * Integration tests for: - {@link CreateDatabaseWithEncryptionKey} - {@link
+ * CreateBackupWithEncryptionKey} - {@link RestoreBackupWithEncryptionKey}
+ */
+@RunWith(JUnit4.class)
+public class EncryptionKeyIT {
+
+  private static String projectId;
+  private static final String instanceId = System.getProperty("spanner.test.instance");
+  private static DatabaseAdminClient databaseAdminClient;
+  private static List<String> databasesToDrop;
+  private static List<String> backupsToDrop;
+  private static Spanner spanner;
+  private static String key;
+
+  @BeforeClass
+  public static void setUp() {
+    final SpannerOptions options = SpannerOptions
+        .newBuilder()
+        .setAutoThrottleAdministrativeRequests()
+        .build();
+    projectId = options.getProjectId();
+    spanner = options.getService();
+    databaseAdminClient = spanner.getDatabaseAdminClient();
+    databasesToDrop = new ArrayList<>();
+    backupsToDrop = new ArrayList<>();
+
+    String keyLocation = System.getProperty("spanner.test.key.location");
+    String keyRing = System.getProperty("spanner.test.key.ring");
+    String keyName = System.getProperty("spanner.test.key.name");
+    key = "projects/" + projectId + "/locations/" + keyLocation + "/keyRings/" + keyRing
+        + "/cryptoKeys/" + keyName;
+  }
+
+  @AfterClass
+  public static void tearDown() {
+    for (String databaseId : databasesToDrop) {
+      try {
+        databaseAdminClient.deleteBackup(instanceId, databaseId);
+      } catch (Exception e) {
+        System.out.println("Failed to drop database " + databaseId + ", skipping...");
+      }
+    }
+    for (String backupId : backupsToDrop) {
+      try {
+        databaseAdminClient.deleteBackup(instanceId, backupId);
+      } catch (Exception e) {
+        System.out.println("Failed to drop backup " + backupId + ", skipping...");
+      }
+    }
+    spanner.close();
+  }
+
+  @Test
+  public void testEncryptedDatabaseAndBackupAndRestore() throws Exception {
+    final String databaseId = DatabaseIdGenerator.generateDatabaseId();
+    final String backupId = DatabaseIdGenerator.generateDatabaseId();
+    final String restoreId = DatabaseIdGenerator.generateDatabaseId();
+
+    databasesToDrop.add(databaseId);
+    backupsToDrop.add(backupId);
+    databasesToDrop.add(restoreId);
+
+    String out = SampleRunner.runSample(() ->
```

Comment:
Have the samples already dropped support for Java 7? Otherwise, this unfortunately won't work....

Reply:
We have only supported java 8 from the start as far as I know (https://github.com/googleapis/java-spanner/blob/master/samples/snippets/pom.xml#L23-L24, https://github.com/googleapis/java-spanner/blob/master/samples/snapshot/pom.xml#L21-L22 and https://github.com/googleapis/java-spanner/blob/master/samples/install-without-bom/pom.xml#L21-L22)